src/robusta/integrations/prometheus/utils.py

```diff
@@ -9,6 +11,53 @@
 from robusta.core.model.env_vars import PROMETHEUS_SSL_ENABLED, SERVICE_CACHE_TTL_SEC
 from robusta.utils.service_discovery import find_service_url
 
+
+class PrometheusAuthorization:
+    bearer_token: str = ""
+    azure_authorization: bool = (
+        os.environ.get("AZURE_CLIENT_ID", "")
+        or os.environ.get("AZURE_CLIENT_ID", "")
+        or os.environ.get("AZURE_CLIENT_SECRET", "")
+    )
+
+    @classmethod
+    def get_authorization_headers(cls, params: PrometheusParams) -> Dict:
+        if params.prometheus_auth:
+            return {"Authorization": params.prometheus_auth.get_secret_value()}
+        elif cls.azure_authorization:
+            return {"Authorization": (f"Bearer {cls.bearer_token}")}
+        else:
+            return {}
+
+    @classmethod
+    def request_new_token(cls) -> bool:
+        if cls.azure_authorization:
+            try:
+                res = requests.post(
+                    url=f"https://login.microsoftonline.com/{os.environ.get('AZURE_TENANT_ID')}/oauth2/token",
```

Comment:
Might be better to allow override for the url and resource, incase they change it and we want a quick workaround (via env var probably)
Just a nice to have